etc/devices/redfishpower-cray-ex.dev

```diff
@@ -0,0 +1,154 @@
+# Support for Redfish Rest Interface
+#
+# How blades and switches in a HPE Cray Supercomputing EX Chassis
+# (abbreviated CrayEX) are populated has a large effect on
+# configuration.  Tweaks to this device file may be needed.  See the
+# section "UPDATING REDFISHPOWER DEVICE FILES" in redfishpower(8) for
+# additional tips.
+#
+# - Set your system's username/password in the login section of
+#   each specification below.
+#
+# - Assuming all blades are populated with nodes and all switches are
+#   populated, configure in Powerman like below.
+#
+#   Take special notice to the order of hosts listed with `-h` as the
+#   order matters (plugnames are mapped to hosts via indices).  The cmm
+#   should be listed first with the 16 nodes second.
+#
+#   include "/etc/powerman/redfishpower-cray-ex.dev"
+#   device "chassis0" "cray-ex" "/usr/sbin/redfishpower -h cmm0,pmynode[0-15] |&"
+#   node "cmm0,myperif[0-7],myblade[0-7],mynode[0-15]" "redfishpower" "Enclosure,Perif[0-7],Blade[0-7],Node[0-15]"
+#
+# - If your chassis is not fully populated, put placeholder hosts in
+#   the redfishpower hosts configuration.  Adjust plugs when
+#   configuring specific targets.  For example, let's say perif[4-7]
+#   and blades[4-7] are not populated (thus nodes[8-15] also do not exist).
+#
+#   device "chassis0" "cray-ex" "/usr/sbin/redfishpower -h cmm0,pnode[0-7],unused[0-7] |&"
+#   node "cmm0,myperif[0-3],myblade[0-3],mynode[0-7]" "redfishpower" "Enclosure,Perif[0-3],Blade[0-3],Node[0-7]"
+#
```

Comment:
This might be slightly clearer if the example focused on the unused elements of the hostlist.  So just depopulate `Blade[4-7]` hence `Node[8-15]` and leave `Perif[4-7]` out of it.

Reply:
Oh one other quick thing.  I assume `auth USER:PASS` is not the published vendor default password.  Do we want to put the default in the dev script ?  Then we wouldn't _require_ people to immediatley copy the dev script. 

It seems like this came up and we found the default on a public web document somewhere?  (need to confirm)

Reply:
Have to check if this is accurate, but if so, it's out there searchable with a google so IMHO is fair game to include in a dev script.

https://support.hpe.com/hpesc/public/docDisplay?docId=a00117940en_us&docLocale=en_US&page=Manage_System_Passwords.html

Reply:
> This might be slightly clearer if the example focused on the unused elements of the hostlist. So just depopulate Blade[4-7] hence Node[8-15] and leave Perif[4-7] out of it.

I guess in my example I wanted to show how to config if switches aren't populated too.  Would it be clearer if I did two examples?

> Oh one other quick thing. I assume auth USER:PASS is not the published vendor default password. Do we want to put the default in the dev script ? Then we wouldn't require people to immediatley copy the dev script.

Good point!

Reply:
> I guess in my example I wanted to show how to config if switches aren't populated too. Would it be clearer if I did two examples?

Eh, I guess if you want although that part is just standard powerman stuff.

Reply:
i'm gonna punt on the default username/password to a follow up PR

Reply:
> Eh, I guess if you want although that part is just standard powerman stuff.

Ohhh, i see your point.  It's normal powerman conf stuff.  makes sense.